datashader/resampling.py

```diff
@@ -27,15 +27,14 @@
 from __future__ import absolute_import, division, print_function
 
 import numpy as np
+import numba as nb
 
 from numba import prange
-from .utils import ngjit, ngjit_parallel
+from .utils import ngjit
 
 try:
-    # Determine if system supports numba 'parallel' target
-    @ngjit_parallel
-    def _sentinel_function(): return 0
-    _sentinel_function()
+    # Try to create numba JIT with 'parallel' target
+    ngjit_parallel = nb.jit(nopython=True, nogil=True, parallel=True)
```

Comment:
Not sure if I am being helpful here (rather than just annoying!) because I am not really following this and I got involved by mistake :) But shouldn't the figuring out of stuff happen in util, where it only needs to happen once?

I.e. in util, you'd do the

```
try:
    ngjit_parallel = nb.jit(parallel=True,...)
    from numba import prange
except:
    ngjit_parallel = ngjit
    prange = range
```

Then all other files wouldn't need to worry about it (they'd just import `ngjit_parallel` and `prange` from util).

I also think you wouldn't need the NOQA this way.

However I might well be missing something. Or it might be bad in that maybe things are less explicit in the files where stuff's actually done. 

Also, finally, maybe this whole approach could result in eventually everyone silently missing out on the parallel stuff when prange is moved within numba, or nb.jit changes in some way, or whatever - anything where stuff changes in the future so that these lines ends up failing for everyone silently and nobody gets parallelized stuff (and worse, nobody knows about it). So maybe at least a warning? And/or maybe catching only whatever error it is you're expecting (`RuntimeError`?). I guess the idea being that for datashader you really don't want to be running without all the optimizations possible, rather than a common opposite requirement of wanting things to at least work even if optimizations are unavailable.

Reply:
True, that seems somewhat cleaner. I think leaving it in resampling is fine for now since ``parallel`` is only needed there. Even without it being parallel it's still 2-3x faster than it was before, so I'm not sure the warning is necessarily needed. I'll leave it up to @jbednar if he wants a warning, if not I'll probably not bother changing it.

Reply:
Those are all good comments.  I'm happy either way; the platforms where we expect parallel to fail are all platforms no one in their right mind should be using, as far as I can see, so I don't mind polluting them with warnings (which would ordinarily be a concern).

Reply:
> the platforms where we expect parallel to fail are all platforms no one in their right mind should be using, as far as I can see

Python 2.7 on 64-bit Windows doesn't seem completely crazy to me.

Reply:
> polluting them with warnings (which would ordinarily be a concern)

I'd like to see some kind of system that warns about all the loud, pink warnings that are going to appear in my notebooks if I choose to use certain packages ;)

Reply:
That's true; Python 2.7 on 64-bit Windows isn't ideal, but isn't crazy; I'm not sure from the ambiguous error message whether that is in the category of platforms not currently supported.